dbt/adapters/databricks/connections.py

```diff
@@ -1147,8 +1384,17 @@ def _get_http_path(node: Optional[ResultNode], creds: DatabricksCredentials) ->
             f"does not specify http_path, relation: {node.relation_name}"
         )
 
-    logger.debug(
-        f"On thread {thread_id}: {node.relation_name} using compute resource '{compute_name}'."
-    )
+    if not USE_LONG_SESSIONS:
+        logger.debug(
+            f"On thread {thread_id}: {node.relation_name} using compute resource '{compute_name}'."
+        )
 
     return http_path
+
+
+def _long_sessions_only(name: str) -> None:
```

Comment:
I feel like it's more pythonic to do something like:

```
assert USE_LONG_SESSIONS, "This path should only be reachable with USE_LONG_SESSIONS"
```

My reasoning is that, I believe your intention is 2-fold:

a.) communicate to readers that these code paths should not be reached if USE_LONG_SESSIONS is False
b.) during development, fail execution if you've violated this pre-condition that you intended to enforce.

By the time it gets to the user, we should not have any execution paths that trigger this exception.  If you are worried that we still might, then I believe the best response is to write more tests to validate that we can't.

Reply:
Long story short, favor asserts for conditions that you intend to enforce, exceptions for things outside of your control.